templates/search/definitionList.tpl

```diff
@@ -1,18 +1,16 @@
 {$categories=$categories|default:true}
-{$displayedUnofficial=false}
-{$displayedSpec=false}
+{$displayedSpec=[]}
 
 {foreach $results as $i => $row}
   {if $categories}
-    {if $row->source->sourceTypeId > 0  && !$displayedSpec[$row->source->sourceTypeId]}
+    {$tid=$row->source->sourceTypeId}
+    {if $tid > 0  && !isset($displayedSpec[$tid])}
       <br>
       <div class="callout callout-info">
-        <h3>{t}{$sourceTypes['label'][$row->source->sourceTypeId]}{/t}</h3>
-        <p class="text-muted">
-          {t}{$sourceTypes['desc'][$row->source->sourceTypeId]}{/t}
-        </p>
+        <h3>{SourceType::getName($tid)}</h3>
```

Comment:
Am evitat pînă acum apelurile de funcții/metode în Smarty.

Reply:
De acord că nu e bine pentru separația logică. Dar am mai făcut asta cu apeluri ieftine care obțin o valoare simplă. Poate și mai bine era să zicem `{$row->source->getTypeName()}`, ca să fie clar că e aproape ca o proprietate, nimic complex. În fine.